### PR TITLE
[FEAT][BREAKING] Upgrade to Glimmer.js 2.0

### PR DESCRIPTION
**Pull request by pzuraq**
This PR upgrades Glimmer Lite to use Glimmer.js 2.0, which is much more
inline implementation-wise. Major changes include:

- Glimmer Component no longer supports `didInsertElement`, `didUpdate`,
  or `bounds`. Instead, users can use modifiers!
- Most custom code removed/upstreamed
- No longer need to add a `key` to `{{each}}`
- Adds support for helper managers and modifier managers, enabling
  custom helpers and modifiers
- `babel-plugin-component-templates` rewritten to produce output that
  feeds into `babel-plugin-strict-precompile`. This is meant to be a
  proto-version of strict-mode templates.

### CODE REVIEW

**Comment by knownasilya** [comment]
Does this need any documentation updates?

File: packages/example-app/package.json [hunk 1]
```diff
@@ -13,12 +13,12 @@
...
   "dependencies": {
-    "glimmer-lite-core": "^0.1.0"
+    "@glimmerx/core": "^0.1.10"
   },
   "devDependencies": {
-    "@glimmer/bundle-compiler": "^0.40.1",
-    "@glimmer/compiler": "^0.40.1",
-    "@glimmer/interfaces": "^0.40.1",
+    "@glimmer/bundle-compiler": "^0.50.0",
+    "@glimmer/compiler": "^0.50.0",
```
Comment (chiragpat) [inline packages/example-app/package.json:20]: I wonder if these two are still needed anymore.

File: packages/@glimmerx/babel-plugin-component-templates/test/fixtures-compat/template-only-component/output.js [hunk 2]
```diff
@@ -1,13 +1,8 @@
-import { setComponentTemplate as _setComponentTemplate } from "@glimmerx/core";
-import _Component from "@glimmerx/component";
+import { templateOnlyComponent as _templateOnlyComponent } from "@glimmer/core";
+import { createTemplate as _createTemplate } from "@glimmer/core";
+import { setComponentTemplate as _setComponentTemplate } from "@glimmer/core";
```
[the remark this answers is not in the document]
Comment (pzuraq) [reply packages/@glimmerx/babel-plugin-component-templates/test/fixtures-compat/template-only-component/output.js:3]: Yeah, this is just the output it creates 😕 I think we can come back to it in the future, it does seem unfortunate, but it should get compiled out anyways in general

**Comment by pzuraq** [comment]
Alright, updated. The new Babel plugin reverts to the previous style, where it can precompile using the new public API from Glimmer.js (exposed from _its_ babel plugin) or compile to `createTemplate()`, which is the compile target for Glimmer.js.

One thing to note is that I had to add a `Program` visitor still to add references to bindings, otherwise by the time we got to the other visitors they would be removed (if they had no other references). I actually checked this on master, and the Example app also has this problem there, so maybe this was a change in Babel recently that wasn't caught by our tests? It removes the reference later, so I think it should be fine.

Also, built the example app using production settings, the result was 119kb, 29.48kb.

**Comment by pzuraq** [comment]
@knownasilya for the most part I don't believe so, most public APIs should be the same.

**Pull request merged**